src/cata_tiles.h

```diff
@@ -1,161 +1,85 @@
 #ifndef CATA_TILES_H
 #define CATA_TILES_H
 
-// make sure that SDL systems are included: Until testing is available for other systems, keep Windows specific
-#if !(defined _WIN32 || defined WINDOWS)
-#include <wordexp.h>
-#endif
-
-#include "SDL2/SDL.h"
-#include "SDL2/SDL_ttf.h"
-
-#include "game.h"
-#include "options.h"
-#include "mapdata.h"
-#include "tile_id_data.h"
-#include "enums.h"
+#include "enums.h"   // point
+#include "weather.h" // weather_printable
 
-#include <map>
-#include <vector>
 #include <string>
+#include <fstream>
+#include <unordered_map>
+#include <vector>
+#include <array>
+
+struct SDL_Renderer;
+struct SDL_Texture;
 
 class JsonObject;
+class Creature;
+class player;
 
-/** Structures */
-struct tile_type
-{
-    int fg, bg;
-    bool multitile, rotates;
+enum class light_type : int;
+enum class tile_category : int;
 
-    std::vector<std::string> available_subtiles;
+struct tile_type {
+    std::string const *id = nullptr;
 
-    tile_type()
-    {
-        fg = bg = 0;
-        multitile = rotates = false;
-        available_subtiles.clear();
-    }
-};
+    int fg = 0;
+    int bg = 0;
 
-struct tile
-{
-    /** Screen coordinates as tile number */
-    int sx, sy;
-    /** World coordinates */
-    int wx, wy;
-
-    tile()
-    {
-        sx = wx = wy = 0;
-    }
-    tile(int x, int y, int x2, int y2)
-    {
-        sx = x;
-        sy = y;
-        wx = x2;
-        wy = y2;
-    }
-};
+    bool rotates      = false;
+    bool multitile    = false;
+    bool has_seasonal = false;
 
-/* Enums */
-enum LIGHTING
-{
-    HIDDEN = -1,
-    CLEAR = 0,
-    LIGHT_NORMAL = 1,
-    LIGHT_DARK = 2,
-    BOOMER_NORMAL = 3,
-    BOOMER_DARK = 4
+    tile_type() = default;
+    tile_type(int const fg, int const bg,
+              bool const rotates = false, bool const multitile = false
+    ) noexcept : fg(fg), bg(bg), rotates(rotates), multitile(multitile) {}
 };
-enum MULTITILE_TYPE
-{
+
+enum multitile_type : int {
     center,
     corner,
     edge,
     t_connection,
     end_piece,
     unconnected,
-    open_,
+    open,
```

Comment:
I though it suspicious and that maybe it had something to do with the stringification going on elsewhere; POSIX didn't really cross my mind. Is it still a problem even if the fully qualified name is used : `multitile_type::open`? If it's a macro I guess it would be a problem. Regardless, a comment would have been nice :+1: